graphql/handler/transport/websocket_test.go

```diff
@@ -265,6 +265,28 @@ func TestWebsocketInitFunc(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, "ok", resp.Empty)
 	})
+
+	t.Run("can set a deadline on a websocket connection and close it with a reason", func(t *testing.T) {
+		h := testserver.New()
+		h.AddTransport(transport.Websocket{
+			InitFunc: func(ctx context.Context, _ transport.InitPayload) (context.Context, error) {
+				newCtx, _ := context.WithTimeout(transport.AppendCloseReason(ctx, "beep boop"), time.Millisecond*5)
+				return newCtx, nil
+			},
+		})
+		srv := httptest.NewServer(h)
+		defer srv.Close()
```

Comment:
```suggestion
		h := testserver.New()
		var cancel context.CancelFunc
		h.AddTransport(transport.Websocket{
			InitFunc: func(ctx context.Context, _ transport.InitPayload) (context.Context, error) {
				var newCtx context.Context
				newCtx, cancel = context.WithTimeout(transport.AppendCloseReason(ctx, "beep boop"), time.Millisecond*5)
				return newCtx, nil
			},
		})
		defer cancel()
		srv := httptest.NewServer(h)
		defer srv.Close()
```
I know it's just for a test, but what do you think about retaining the closeFunc and calling it in a defer somewhere to satisfy the linter that there's no leaks?